src/csharp/Adapters.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.ML.OnnxRuntimeGenAI;
+using System;
+using System.Runtime.InteropServices;
+
+/// <summary>
+/// A container of adapters.
+/// </summary>
+public class Adapters : SafeHandle
+{
+    private Adapters(IntPtr handle) :
+        base(handle, true)
+    {
+    }
+
+    /// <summary>
+    /// Creates a container for adapters
+    /// used to load, unload and hold them.
+    /// Throws on error.
+    /// </summary>
+    /// <param name="model">Reference to a loaded model</param>
+    /// <returns>new Adapters object</returns>
+    public static Adapters Create(Model model)
+    {
+        Result.VerifySuccess(NativeMethods.OgaCreateAdapters(model.Handle, out IntPtr handle));
+        return new Adapters(handle);
+    }
```

Comment:
Should we use the constructor instead of a static `Create` method?

```suggestion
    /// <summary>
    /// Creates a container for adapters
    /// used to load, unload and hold them.
    /// Throws on error.
    /// </summary>
    /// <param name="model">Reference to a loaded model</param>
    /// <returns>new Adapters object</returns>
    public Adapters(Model model)
        {
            Result.VerifySuccess(NativeMethods.OgaCreateAdapters(model.Handle, out _handle));
        }
```

Reply:
One issue I am facing is that C# API is not consistent with C++. It tends to use constructors instead of static methods like on C++. I think this should be consistent.

Reply:
The reason for the C++ API needing the Create method is because it essentially does not hold any state and is used as a pass-through object with the same name as the underlying C opaque object.
C# does not have any such problems it needs to address. And so, I think it would be ok to make C# and C++ api inconsistent. Considering other API use constructors, I would be in favor of using the Adapters constructor instead of a dedicated static Create method.